src/LanguageServer.SemanticModel.Xml/XSPath.cs

```diff
@@ -29,8 +29,8 @@ public sealed class XSPath // TODO: IEquitable<XSPath> (plus equality / inequali
         ///     The root path.
         /// </summary>
         public static readonly XSPath Root = new XSPath(
-            ancestorSegments: ImmutableList<XSPathSegment>.Empty,
-            segments: ImmutableList<XSPathSegment>.Empty.Add(XSPathSegment.Root)
+            ancestorSegments: [],
```

Comment:
I am not sure think this does what it appears to suggest - the immutable collection types may not play nicely with collection expressions unless they are smart enough to work out that AddRange on an empty sequence is equivalent to .Empty.

Reply:
It’s probably not that big a deal BTW (it’s probably creating an empty array and then passing that to a new immutable list).

Reply:
https://stackoverflow.com/a/77623157

Reply:
Collection expressions are designed to only guarantee the resulting collection, but not the way it was constructed. Yes, currently this case produces a bit worse underlying IL, but as time goes sich cases are gonna be optimized. E.g. this one is already tracked in roslyn: https://github.com/dotnet/roslyn/issues/71159. So let's preffer readability here now since it isn't some kind of hot path and the codegen is gonna be improved later in newer versions of the compiler

Reply:
Agree 100%